### PR TITLE
Issue826-IntegrateEventFrameworkIntoALS

### PR DESCRIPTION
**Pull request by gibaros**
Changes:

-Register event plugin
-update swagger operation ids to follow convention
-configure event framework to all implemented operations
-update unit tests to account for headers and span parameters
-run audit check and resolve and set reminder
-run dependency check and update
-merged latest MASTER into development branch as of Dec 10th, 2019
-set package version to 8.6.3

### CODE REVIEW

**Comment by gibaros** [comment]
> Hi @gibaros, thanks for the introduced changes
> 
> * I could successfully start the service with and without the event-sidecar
> * unit tests pass
> * integration tests pass
> * audits are showing on the dashboard
> 
> The PR looks pretty good! You will need to bring the branch up to date.
> 
> There's just one small change, that is needed, imo. Please, see the attached screenshot:
> image
> I believe, there should not be tags with undefined values, thus you might need to change getSpanTags to omit those. @mdebarros to confirm

Thank you @ggrg for the review! Hi @mdebarros per comment should I go ahead and remove tags showing undefined values?

File: src/handlers/oracles.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports = {
    * responses: 200, 400, 401, 403, 404, 405, 406, 501, 503
    */
   get: async (request, h) => {
-    const response = await oracle.getOracle(request.query)
+    const response = await oracle.getOracle(request.query, request.headers, request.span)
```
Comment (mdebarros) [inline src/handlers/oracles.js]: Normally this is where we do the initial span.audit with action EventSdk.AuditEventAction.start --- example --> https://github.com/mojaloop/ml-api-adapter/blob/master/src/api/transfers/handler.js

File: src/handlers/oracles.js [hunk 2]
```diff
@@ -49,7 +49,7 @@ module.exports = {
    * responses: 201, 400, 401, 403, 404, 405, 406, 501, 503
    */
   post: async (request, h) => {
-    await oracle.createOracle(request.payload)
+    await oracle.createOracle(request.payload, request.headers, request.span)
```
Comment (mdebarros) [inline src/handlers/oracles.js]: Normally this is where we do the initial span.audit with action EventSdk.AuditEventAction.start --- example --> https://github.com/mojaloop/ml-api-adapter/blob/master/src/api/transfers/handler.js

File: src/handlers/oracles.js (hunk 1 above)
Comment (mdebarros) [inline src/handlers/oracles.js]: Missing nitial span.audit with action EventSdk.AuditEventAction.start --- example --> https://github.com/mojaloop/ml-api-adapter/blob/master/src/api/transfers/handler.js

File: src/handlers/oracles.js (hunk 2 above)
Comment (mdebarros) [inline src/handlers/oracles.js]: Missing nitial span.audit with action EventSdk.AuditEventAction.start --- example --> https://github.com/mojaloop/ml-api-adapter/blob/master/src/api/transfers/handler.js

File: src/handlers/oracles.js (hunk 1 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/domain/oracle/oracle.js getOracle function. Please let me know if needs to come up at the src/handlers/oracles.js level.

File: src/handlers/oracles.js (hunk 2 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/domain/oracle/oracle.js createOracle function. Please let me know if needs to come up at the src/handlers/oracles.js level.

Comment (gibaros) [reply src/handlers/oracles.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/domain/oracle/oracle.js createOracle function. Please let me know if needs to come up at the src/handlers/oracles.js level.

Comment (gibaros) [reply src/handlers/oracles.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/domain/oracle/oracle.js createOracle function. Please let me know if needs to come up at the src/handlers/oracles.js level.

File: src/handlers/oracles.js (hunk 1 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/domain/oracle/oracle.js getOracle function. Please let me know if needs to come up at the src/handlers/oracles.js level.

File: src/domain/participants/participants.js [hunk 3]
```diff
@@ -217,17 +218,17 @@ const putParticipantsErrorByTypeAndID = async (headers, params, payload, dataUri
  * @param {string} method - http request method
  * @param {object} params - uri parameters of the http request
  * @param {object} payload - payload of the request being sent out
- *
+ * @param {object} span
  */
-const postParticipants = async (headers, method, params, payload) => {
+const postParticipants = async (headers, method, params, payload, span) => {
   try {
```
[the remark this answers is not in the document]
Comment (gibaros) [reply src/domain/participants/participants.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/handlers/participants.js POST method. Please let me know if needs to come down at the src/domain/participants/participants.js level.

File: src/domain/participants/participants.js [hunk 4]
```diff
@@ -42,17 +43,17 @@ const Config = require('../../lib/config')
  * @param {object} params - uri parameters of the http request
  * @param {string} method - http request method
  * @param {object} query - uri query parameters of the http request
- *
+ * @param {object} span
  * */
-const getParticipantsByTypeAndID = async (headers, params, method, query) => {
+const getParticipantsByTypeAndID = async (headers, params, method, query, span) => {
   try {
```
[the remark this answers is not in the document]
Comment (gibaros) [reply src/domain/participants/participants.js]: Initial span.audit with action EventSdk.AuditEventAction.start is at src/handlers/participants/{Type}/{ID}.js GET method. Please let me know if needs to come down at the src/domain/participants/participants.js level.

File: src/handlers/oracles.js (hunk 1 above)
Comment (mdebarros) [reply src/handlers/oracles.js]: Yes, please. It should be consistent and handled in the handlers directly.

File: src/handlers/oracles.js (hunk 2 above)
Comment (mdebarros) [reply src/handlers/oracles.js]: Same as above.

File: src/handlers/oracles.js (hunk 1 above)
Comment (mdebarros) [reply src/handlers/oracles.js]: Same as above.

File: src/handlers/oracles.js (hunk 2 above)
Comment (mdebarros) [reply src/handlers/oracles.js]: Same as above.

File: src/domain/participants/participants.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (mdebarros) [reply src/domain/participants/participants.js]: That is fine. Lets keep in the handlers.

File: src/domain/participants/participants.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdebarros) [reply src/domain/participants/participants.js]: That is fine. Lets keep in the handlers.

File: src/handlers/oracles.js (hunk 2 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Code has been refactored to have all newly added event sdk code at handlers level.

File: src/handlers/oracles.js (hunk 1 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Code has been refactored to have all newly added event sdk code at handlers level.

File: src/handlers/oracles.js (hunk 2 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Code has been refactored to have all newly added event sdk code at handlers level.

File: src/handlers/oracles.js (hunk 1 above)
Comment (gibaros) [reply src/handlers/oracles.js]: Code has been refactored to have all newly added event sdk code at handlers level.

File: src/domain/participants/participants.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (gibaros) [reply src/domain/participants/participants.js]: Code has been refactored to have all newly added event sdk code at handlers level.

File: src/domain/participants/participants.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (gibaros) [reply src/domain/participants/participants.js]: Code has been refactored to have all newly added event sdk code at handlers level.

**Comment by gibaros** [comment]
Code has been refactored to have all newly added event sdk code at handlers level. PR ready for final review.

**Pull request merged**